src/components/hooks/useActivityStream.js

```diff
@@ -2,52 +2,57 @@ import {useContext, useEffect, useReducer} from 'react';
 
 import {AdapterContext} from '../../components/';
 
-const APPEND_ACTIVITIES = 'append_activities';
+export const PREPEND_ACTIVITIES = 'prepend_activities';
+export const APPEND_ACTIVITIES = 'append_activities';
 
 /**
- * Returns a new array of activityIDs based on the given action.
- * If no action is passed, it will return the same activityIDs
+ * Returns a new array of activities based on the given action.
+ * If no action is passed, it will return the same activities
  * array without any changes.
  *
- * @param {Array.<string>} activityIDs activityIDs associated to the room
- * @param {object} action action to apply to given activityIDs
- * @returns {Array.<string>} New activityIDs associated to the room
+ * @param {Array.<string|ActivityDate>} activities activities associated to the room
+ * @param {object} action action to apply to given activities
+ * @returns {Array.<string|ActivityDate>}
  */
-function reducer(activityIDs, action) {
-  let newActivityIDs = [];
+function reducer(activities, action) {
+  let newActivities = [];
 
   switch (action.type) {
+    case PREPEND_ACTIVITIES:
+      newActivities = action.payload.reverse().concat(activities);
```

Comment:
Why do we need to reverse the order of the previous activities? Should the adapter return them in proper order?

Reply:
My assumption is that most data retrieved is sorted most-recent to older. Also, displaying the data in reverse order is a function of the UX of the activity stream, and not so much of the data itself, and so, part of the UI logic.
But maybe that's not the case on how data gets sent from the backend?

Reply:
Just investigated and saw that the `/messages` endpoint returns data in reverse date sort (latest to oldest). Conversation endpoint does the opposite (oldest to latest). If we aren't going to do any kind of "sorting" in this hook, we need to make sure to document the expected order. 

We should expect the correct order from the adapter and reduce the code here to just prepend the array to the existing array.

Reply:
My initial instinct is also to have activities sorted newest-to-oldest, as we on the `/messages` API; second reason being consistent with public APIs. However, I've changed and documented the expected order as you requested.